code/modules/jobs/job_types/mentor/trainingofficer.dm

```diff
@@ -39,26 +39,67 @@
 		/obj/item/melee/classic_baton,
 		/obj/item/info_printer,
 		/obj/item/announcementmaker/lcorp,
-		/obj/item/suppressionupdate/training_officer,
+		/obj/item/stat_equalizer,
 		/obj/item/sensor_device,
 	)
 
 	//Training Stat update
-/obj/item/suppressionupdate/training_officer
+/obj/item/stat_equalizer
 	name = "training stat equalizer"
 	desc = "A localized source of stats, only usable by the Training Officers to equalize the stats of them and any interns."
 	icon_state = "records_stats"
-	allowedroles = list("Records Agent")
 
-/obj/item/suppressionupdate/training_officer/attack_self(mob/living/carbon/human/user)
-	if(!LAZYLEN(allowedroles))
-		if(!istype(user) || !(user?.mind?.assigned_role in allowedroles))
-			to_chat(user, span_notice("The Gadget's light flashes red. You aren't a Training Officer. Check the label before use."))
-			return
+/obj/item/stat_equalizer/attack_self(mob/living/carbon/human/user)
+	if(!istype(user) || user?.mind?.assigned_role != "Records Agent")
+		to_chat(user, span_notice("The Gadget's light flashes red. You aren't a Training Officer. Check the label before use."))
+		return
 	update_stats(user)
 
-/obj/item/suppressionupdate/training_officer/attack(mob/living/M, mob/user)
-	if(!istype(user) || !(user?.mind?.assigned_role in allowedroles))
+/obj/item/stat_equalizer/attack(mob/living/M, mob/user)
+	if(!istype(user) || user?.mind?.assigned_role != "Records Agent")
 		to_chat(user, span_notice("The Gadget's light flashes red. You aren't a Training Officer. Check the label before use."))
 		return
 	update_stats(M)
+
+/obj/item/stat_equalizer/proc/update_stats(mob/living/carbon/human/user)
+	var/list/attribute_list = list(FORTITUDE_ATTRIBUTE, PRUDENCE_ATTRIBUTE, TEMPERANCE_ATTRIBUTE, JUSTICE_ATTRIBUTE)
+
+	//I got lazy and this needs to be shipped out today
+	var/set_attribute = 20
+	var/facility_full_percentage = 0
+	if(SSabnormality_queue.spawned_abnos) // dont divide by 0
+		facility_full_percentage = 100 * (SSabnormality_queue.spawned_abnos / SSabnormality_queue.rooms_start)
+	// how full the facility is, from 0 abnormalities out of 24 cells being 0% and 24/24 cells being 100%
+	switch(facility_full_percentage)
+		if(15 to 29) // Shouldn't be anything more than TETHs (4 Abnormalities)
+			set_attribute *= 1.5
+
+		if(29 to 44) // HEs (8 Abnormalities)
+			set_attribute *= 2
+
+		if(44 to 59) // A bit before WAWs (11 Abnormalities)
+			set_attribute *= 2.5
+
+		if(59 to 69) // WAWs around here (15 Abnormalities)
+			set_attribute *= 3
+
+		if(69 to 79) // ALEPHs starting to spawn (17 Abnormalities)
+			set_attribute *= 3.5
+
+		if(79 to 100) // ALEPHs around here (20 Abnormalities)
+			set_attribute *= 4
+
+	if(user?.mind?.assigned_role == "Records Agent") //only give these if you're the guy that's allowed to have this, it's here so the Rec Agent can use it on interns
+		set_attribute += GetFacilityUpgradeValue(UPGRADE_AGENT_STATS) + SSlobotomy_corp.ordeal_stats
+
+	else
+		return
```

Comment:
Reasonably you can't even reach here without being a "Records Agent" due to previous checks, but this check should be higher to prevent redundant operations. Also inverted as part of that to simply be
```
if(user?.mind?.assigned_role != "Records Agent")
	return
```